### ISSUE TITLE
Rule Change: (no-restricted-globals) Add whitelist option instead of blacklist

### ISSUE THREAD

**Issue by ST-DDT** [eslint/eslint#19024]
### What rule do you want to change?

[no-restricted-globals](https://eslint.org/docs/latest/rules/no-restricted-globals)

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

Buffer.from(data).toString('base64');

### What does the rule currently do for this code?

Currently the rule needs to be configured in a way that disallows all globals that exists somewhere but potentially not in all target platforms.

e.g. Buffer is available in Node, but not in browsers without polyfills.

This is currently possible via this workaround:

````ts
import globals from 'globals';

const browserGlobals = new Set(Object.keys(globals.browser));
const disallowedGlobals = Object.keys(globals.node).filter(
  (e) => !browserGlobals.has(e)
);

...
'no-restricted-globals': ['error', 'Intl', ...disallowedGlobals],
````

But that is kind of ugly.

### What will the rule do after it's changed?

The rule would have an option that would allow listing exactly which globals are acceptable to use.

e.g.

````ts
'no-restricted-globals': ['error', {allowed: ['Math', 'Number', 'JSON'] } ],
````

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Related Issue:

- https://github.com/faker-js/faker/issues/3145

**Comment by mdjermanovic** [comment]
> The rule would have an option that would allow listing exactly which globals are acceptable to use.

I think this can already be achieved by specifying only the globals you want to allow in [`languageOptions.globals`](https://eslint.org/docs/latest/use/configure/language-options#using-configuration-files). Then, the `no-undef` rule would report on attempts to use others. Would that work for your use case?

**Comment by ST-DDT** [comment]
Yeah, that works fine, however I had to configure it at the end because typescript-eslint's recommended rule turns it off.
